frappe/__init__.py

```diff
@@ -293,11 +289,22 @@ def connect(
 			"Instead, explicitly invoke frappe.init(site) prior to calling frappe.connect(), if initializing the site is necessary."
 		)
 		init(site)
+	if db_name:
+		from frappe.utils.deprecations import deprecation_warning
+
+		deprecation_warning(
+			"Calling frappe.connect with the db_name argument is deprecated and will be removed in next major version. "
+			"Instead, explicitly invoke frappe.init(site) with the right config prior to calling frappe.connect(), if necessary."
+		)
+
+	assert db_name or local.conf.db_user, "site must be fully initialized, db_user missing"
```

Comment:
Shouldn't we swap the order (both here and on line 307) - if someone has a user sent in the configuration that should be prioritized?

Reply:
Given that this removes the only known usage of this argument and deprecates it, I think you're right. That is a saner approach and only leaves this as escape hatch rather than an override.

I will change it shortly.

Reply:
Ok, it's done: https://github.com/frappe/frappe/pull/24432/commits/161bb4250871d23c346b40f1640562b054089c8b